test/models/indicator_test.rb

```diff
@@ -0,0 +1,130 @@
+require 'test_helper'
+
+class IndicatorTest < ActiveSupport::TestCase
+  # Test Relationships
+  should belong_to(:level)
+  should belong_to(:competency)
+
+
+  # Basic Validations
+  should validate_presence_of(:description)
+  should validate_presence_of(:level_id)
+  should validate_presence_of(:competency_id)
+  should validate_numericality_of(:level_id)
+  should validate_numericality_of(:competency_id)
+
+  should allow_value("Indicator 1").for(:description)
+  should_not allow_value("").for(:description)
+
+  should allow_value(1).for(:level_id)
+  should_not allow_value(5.3).for(:level_id)
+  should_not allow_value("").for(:level_id)
+  should_not allow_value("id").for(:level_id)
+
+  should allow_value(1).for(:competency_id)
+  should_not allow_value("").for(:competency_id)
+  should_not allow_value("id").for(:competency_id)
+  should_not allow_value(7.5).for(:competency_id)
+
+
+  # Scope and method tests
+  context "With a proper context, " do
+    setup do
+      create_competencies
+      create_levels
+      create_indicators
+    end
+
+    teardown do
+      remove_competencies
+      remove_levels
+      remove_indicators
+    end
+
+    # Test that objects are created properly
+    should "show that all factory objects are properly created" do
+      assert_equal false, @indicator1.nil?
+      assert_equal "Able to identify common nonverbal cues.", @indicator1.description
+      assert_equal "Able to present written communication in an easy–to-read format.", @indicator2.description
+      assert_equal "Communication", @indicator3.competency.name
+      assert_equal "Companion", @indicator4.level.name
+      assert_equal false, @indicator5.active
+    end
+
+    # test alphabetical scope
```

Comment:
So I think we should add it for now because our default sort should be alphabetical. Like if we sorted all indicators by level, we should sort all indicators within each level alphabetically.